src/cpp/middleware/fastdds/FastDDSMiddleware.cpp

```diff
@@ -566,6 +566,9 @@ bool FastDDSMiddleware::create_requester_by_bin(
         attrs.subscriber.topic.topicName = requester_xrce.reply_topic_name();
         attrs.subscriber.topic.topicDataType = requester_xrce.reply_type();
 
+        attrs.publisher.historyMemoryPolicy = fastrtps::rtps::PREALLOCATED_WITH_REALLOC_MEMORY_MODE;
```

Comment:
Does this apply to `create_replier_by_bin`?

Reply:
Yes, I'm adding it

Reply:
ad816be953f30cd2459858aa4c0d940027c175a3